File: Tests/Reqnroll.Formatters.Tests/Samples/Resources/global-hooks-attachments/global_hooks_attachments.ts
import { When, BeforeAll, AfterAll } from '@cucumber/fake-cucumber'

BeforeAll({}, async function () {
  await this.attach('Attachment from BeforeAll hook', 'text/plain')
})

When('a step passes', function () {
  // no-op
})

AfterAll({}, async function () {
  await this.attach('Attachment from AfterAll hook', 'text/plain')
})

File: Tests/Reqnroll.Formatters.Tests/Samples/Resources/global-hooks-attachments/global_hooks_attachments.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Reqnroll;

namespace Reqnroll.Formatters.Tests.Samples.Resources.global_hooks_attachments;

[Binding]
internal class global_hooks_attachments
{

    [BeforeTestRun]
    public static void BeforeAllHook(IReqnrollOutputHelper reqnrollOutputHelper)
    {
        reqnrollOutputHelper.WriteLine("Attachment from BeforeAll hook");
    }

    [When("a step passes")]
    public void WhenAStepPasses() { }

    [AfterTestRun]
    public static void AfterAllHook(IReqnrollOutputHelper reqnrollOutputHelper)
    {
        reqnrollOutputHelper.WriteLine("Attachment from AfterAll hook");
    }
}
